cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/RemoteDatasetFramework.java

```diff
@@ -86,6 +91,12 @@ public DatasetServiceClient load(Id.Namespace namespace) throws Exception {
 
   @Override
   public void addModule(Id.DatasetModule moduleId, DatasetModule module) throws DatasetManagementException {
+    addModule(moduleId, module, null);
```

Comment:
This is called when trying to add system dataset module or when the JAR is not available (it will not be the case in current code base). Added javadoc in `DatasetFramework` to explain that.